### PR TITLE
CSRF: Moved from be.net

### PR DESCRIPTION
**Pull request by jrefano**

### CODE REVIEW

File: util/xhr.js [hunk 1]
```diff
@@ -1,15 +1,42 @@
...
-        req = $.ajax.apply($, arguments);
+        req;
+
+    if (typeof options === 'string') {
+      options = { url: options };
+    }
+
+    options = addCsrfToken(options);
 
+    req = $.ajax(options);
```
Comment (mikesherov) [inline util/xhr.js]: can't do this. It's almost right, but it's possible to call `$.ajax` with `$.ajax(url, options);` so you have to account for that possibility.

File: util/xhr.js [hunk 2]
```diff
@@ -1,15 +1,42 @@
...
   * This is so we can get proper error reporting
   */
-  return function xhr() {
+  return function xhr(options) {
     var thenable,
         p = new Promise(),
-        req = $.ajax.apply($, arguments);
+        req;
+
+    if (typeof options === 'string') {
```
Comment (mikesherov) [inline util/xhr.js]: oh, no, you did the right thing!

**Comment by mikesherov** [comment]
👍

**Comment by mmase** [comment]
:+1:

**Pull request merged**